plugins/fabrik_element/cascadingdropdown/cascadingdropdown.php

```diff
@@ -824,10 +824,26 @@ protected function buildQuery($data = array(), $incWhere = true, $opts = array()
 			{
 				foreach ($whereval as &$v)
 				{
-					$v = $db->quote($v);
-				}
 
-				$where .= count($whereval) == 0 ? '1 = -1' : $wherekey . ' IN (' . implode(',', $whereval) . ')';
+					// Jaanus: Solving bug: imploded arrays when chbx in repeated group         
+					
+					if (is_array($v)) 
+					{
+						foreach ($v as &$vchild)
+						{
+							$vchild = FabrikString::safeQuote($vchild);
+						}
+						$v = implode(',', $v);
+					}
+					else
+					{
+						$v = FabrikString::safeQuote($v);
+					}
+				}
+      
+				// Jaanus: if count of where values is 0 or if there are no letters or numbers, only commas in imploded array
+				
+				$where .= count($whereval) == 0 || !preg_match('/\w/', implode(',', $whereval)) ? '1 = -1' : $wherekey . ' IN ' . '(' . str_replace(',,', ',\'\',', implode(',', $whereval)) . ')';
```

Comment:
can be simplified - as $v will either be a string or an array which you can count, no need to do regular expression match on the imploded string